baybe/campaign.py

```diff
@@ -58,24 +60,42 @@ class Campaign(SerialMixin):
     strategy: Strategy = field(factory=TwoPhaseStrategy)
     """The employed strategy"""
 
-    # Data
-    measurements_exp: pd.DataFrame = field(factory=pd.DataFrame, eq=eq_dataframe)
-    """The experimental representation of the conducted experiments."""
-
-    numerical_measurements_must_be_within_tolerance: bool = field(default=True)
-    """Flag for forcing numerical measurements to be within tolerance."""
-
     # Metadata
-    n_batches_done: int = field(default=0)
+    n_batches_done: int = field(default=0, init=False)
     """The number of already processed batches."""
 
-    n_fits_done: int = field(default=0)
+    n_fits_done: int = field(default=0, init=False)
     """The number of fits already done."""
 
     # Private
-    _cached_recommendation: pd.DataFrame = field(factory=pd.DataFrame, eq=eq_dataframe)
+    _measurements_exp: pd.DataFrame = field(
+        factory=pd.DataFrame, eq=eq_dataframe, init=False
+    )
+    """The experimental representation of the conducted experiments."""
+
+    _cached_recommendation: pd.DataFrame = field(
+        factory=pd.DataFrame, eq=eq_dataframe, init=False
+    )
     """The cached recommendations."""
 
+    # Deprecation
+    numerical_measurements_must_be_within_tolerance: bool = field(default=None)
+    """Deprecated! Raises an error when used."""
+
+    @numerical_measurements_must_be_within_tolerance.validator
+    def _validate_tolerance_flag(self, _, value) -> None:
+        if value is not None:
+            raise DeprecationError(
+                f"Passing a 'numerical_measurements_must_be_within_tolerance' flag to "
+                f"the constructor is deprecated. The flag has become a parameter of "
+                f"{self.__class__.__name__}.{Campaign.add_measurements.__name__}."
+            )
+
+    @property
+    def data(self) -> pd.DataFrame:
```

Comment:
I know we agreed, but Im getting second thoughts on renaming this. Rason: our emthod for adding data is called `add_measurements` and not `add_data`. It would be nice to retain the connection between these two objects via the name

Reply:
Yes, I noticed the same thing. That's why I wanted to hear your thoughts on whether we should
* stay close to what we have and call the property `measurements` (still, I find the name somewhat confusing for a user) or
* we should rename the other one to `add_data`
* ... (other options)

Reply:
so for me it doenst sound strange (in our language we have `targets` to express what you mean)

but if the issue s strong enough for you then we should change to `data`. But if its not major then we should retain `measurements` as it doesnt change on of the three big userfacing methods

Reply:
Have no strong opinion here. Let's ask @AVHopp , and maybe in particular @RimRihana, since she's a new user?

Reply:
Well, for me measurements sound more specific than data, therefore it is easier to understand the context :)

Reply:
I'm also worried that data might be too general. For me, it would be not that clear what `campaign.data` returns, since the term `data` is just too vague. So I'd prefer `measurements`. Also, if we want to allow a more complex name, we could also have `measured_experiments` or something like that? Although I am not sure whether the term "experiment" maybe adds more confusion than clarification 😆

Reply:
well thats a clear vote, I assume you change it back and approve the PR 4unblock

Reply:
OK, agree. Will change it to `measurements` then (at least we got rid of the "_exp" suffix for the public attribute, which is nothing a user should bother with) 👍🏼